### DISCUSSION TITLE
Instancing GlMeshItem

### DISCUSSION THREAD

**Discussion by visar77** [pyqtgraph/pyqtgraph#2590] [Q&A]
I'm trying to draw thousands of GLMeshItems, they all have the same MeshData and only differ in their position and maybe color. Furthermore, I am updating the Plot and adding 20 GLMeshItem per second at different positions. Does anybody know or have already implemented Instancing or VBOs so that the plot doesn't lag and the perfomance doesn't drop immensely. 

My idea is to completely override the paint() method, but I don't have any idea how to implement instancing with pyopengl and using OpenGL Instancing examples is kinda hard, because it's hard to translate cpp code to python, even if the methods are the same.

**Comment by j9ac9k** [comment]
I'm not as familiar with the 3D aspects of our library, but thousands of GLMeshItems would likely be problematic, even if they are identical in features except for positions.  @campagnola may have thoughts on how to speed this up, but with that kind of size, VisPy might be better suited.

Truth be told that the 3D elements of our library have been mostly on a bug-fix and compatibility type changes, as none of the current active maintainers have particular expertise with opengl.  

Our general recommendation is to try and minimize the number of items you're trying to plot, but I know that's not always feasible.
